File: 118-pascals-triangle/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> sol = new ArrayList<>();
       
        for (int i = 0; i < numRows; i++) {
            if (i == 0) { 
                List<Integer> row = new ArrayList<>();
                row.add(1);
                sol.add(row);
            }
            else if (i == 0) { 
                List<Integer> row = new ArrayList<>();
                row.add(1); row.add(1);
                sol.add(row);
            }
            else {
                List<Integer> row = new ArrayList<>();
                row.add(1);
                List<Integer> prev = sol.get(sol.size() - 1);
                for (int j = 1; j < prev.size(); j++) {
                    row.add(prev.get(j - 1) + prev.get(j));
                }
                row.add(1);
                sol.add(row);

            }
            
        }
        return sol;
    }
}

File: 118-pascals-triangle/pascals-triangle.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        bigList = []
        for i in range(numRows): 
            if (i == 0):
                bigList.append([1]) 
            elif (i == 1):
                bigList.append([1,1])     
            else:
                prev = bigList[-1]
                row = [1]
                for j in range(1, len(prev)):
                    row.append(prev[j - 1] + prev[j])
                row.append(1)
                bigList.append(row)
        return bigList
                 
           
        
        
            

       
        